sw/device/silicon_creator/manuf/lib/BUILD

```
@@ -283,3 +285,56 @@ opentitan_functest(
         "//sw/device/silicon_creator/lib/drivers:otp",
     ],
 )
+
+opentitan_ram_binary(
+    name = "sram_exec_test",
+    srcs = [
+        "sram_exec_test.c",
+    ],
+    archive_symbol_prefix = "sram_exec_test",
+    deps = [
+        "//sw/device/examples/sram_program:sram_program_linker_script",
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/lib/arch:device",
+        "//sw/device/lib/base:macros",
+        "//sw/device/lib/runtime:log",
+        "//sw/device/lib/testing:otp_ctrl_testutils",
+        "//sw/device/lib/testing:pinmux_testutils",
+        "//sw/device/lib/testing/test_framework:check",
+        "//sw/device/lib/testing/test_framework:status",
+        "//sw/device/lib/dif:otp_ctrl"
+    ],
+)
+
+# We are using a bitstream with disabled execution so the content of the flash
+# does not matter but opentitan_functest() is unhappy if we don't provide one.
+# Since execution in the ROM is disabled, bootstrap is not possible so we need
+# to make sure that the test does not try to bootstrap
+opentitan_functest(
+    name = "manuf_cp_ast_test_execution",
+    srcs = ["empty_functest.c"],
+    cw310 = cw310_params(
+        bitstream = "//hw/bitstream:rom_otp_dev_initial",
+        tags = ["cw310_rom"],
+        test_cmds = [
+            "--rom-kind=rom",
+            "--bitstream=\"$(rootpath {bitstream})\"",
+            "--vmem=\"$(rootpath :sram_exec_test_fpga_cw310_vmem)\"",
+            # the following values come from the sram linker script
+            "--load-addr=0x10001fc4",
+            "--stack-addr=0x10020000",
+            "--stack-size=1024",
+            "--global-pointer=0x100027C4", # sram_load_addr + 2048;
+        ] + OPENTITANTOOL_OPENOCD_TEST_CMDS,
+    ),
+    data = [
+        ":sram_exec_test_fpga_cw310_vmem",
+    ] + OPENTITANTOOL_OPENOCD_DATA_DEPS,
+    targets = ["cw310_rom"],
+    test_harness = "//sw/host/tests/manuf/manuf_cp_ast_test_execution:manuf_cp_ast_test_execution",
```

pro-tip, some [bazel shorthand](https://bazel.build/concepts/labels) when the package name matches the target name in the label you can omit the target name:
```suggestion
    test_harness = "//sw/host/tests/manuf/manuf_cp_ast_test_execution",
```